### ISSUE TITLE
Unhandled Exception: FormatException: Missing extension byte (at offset 42)

### ISSUE THREAD

**Issue by filipenanclarez** [cbl-dart/cbl-dart#371]
**Describe the bug**

When the data for sync is big, then the replicator get this exception.

**Environment:**

 - OS: Windows, Android and iOS
 - `cbl`: 1.2.0
 - `cbl_flutter`: 1.0.0+3
 - `cbl_flutter_ce`: 1.1.1

**Screenshots**

Here the problem:

![image](https://user-images.githubusercontent.com/27212546/168605573-2ccfdee4-6221-4b8d-8bb8-87c1b814a410.png)

This the moment of exception, watching the variables

![image](https://user-images.githubusercontent.com/27212546/168604632-bc4be99c-d64e-4f05-a97b-894ff82dd45a.png)

Here the value of `arguments[2]`
 
![image](https://user-images.githubusercontent.com/27212546/168604360-5499749a-b8fa-4db7-8e95-98d96b2ccf25.png)

**Comment by blaugold** [comment]
Thanks for reporting this.

There are two separate issues here.

Firstly, the exceptions you are seeing are caused by a log messages not being correctly encoded as UTF-8.
We get this log message from the underlying Couchbase Lite C SDK. Either the SDK is not handling UTF-8 correctly somewhere or is receiving some badly encoded text from the network.
We can relatively easily workaround this issue by allowing malformed characters when decoding the log message. I'll create a fix for that.
When decoding the log message and allowing malformed characters, this is what it says:
```
ClientSocket got POSIX error 108 "Foi for�ado o nelamento de uma conex�o existente pelo host remoto."
```

From that message, we can see that the connection to the server was reset. This is a known issue, that will be fixed in an upcoming version of Couchbase Lite C, but should not prevent replication because the replicator retries to replicate documents when it sees this error.

**Comment by blaugold** [comment]
Fixed by #372
